src/view/hooks/useUploadImage.ts

```diff
@@ -0,0 +1,76 @@
+import { getApp } from "firebase/app";
+import {
+    getDownloadURL,
+    getStorage,
+    ref,
+    uploadBytesResumable,
+    UploadTask,
+} from "firebase/storage";
+import { useState } from "react";
+
+const useUploadImage = () => {
+    const [file, setFile] = useState<File | null>(null);
+    const [imageURL, setImageURL] = useState<string>("");
```

Comment:
[refactor] 値が入っているのと入っていない状態を明確に区別するために、こうしたほうが良さそうです。
```suggestion
    const [imageURL, setImageURL] = useState<string|null>(null);
```